src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -850,12 +856,10 @@ def build_and_upload_locally(src_dir, mode, overwrite=False, archive=False, publ
             projects_by_region = {region: dest_project}
 
             if not overwrite and not archive:
-                # If we cannot overwite or archive an existing applet and an
+                # If we cannot overwrite or archive an existing applet and an
                 # applet in the destination exists with the same name as this
                 # one, then we should err out *before* uploading resources.
                 try:
-                    if not override_applet_name and kwargs.get("name"):
```

Comment:
This check is no longer needed as we are dealing with everything when assigning value to `override_applet_name `